modules/mono/glue/GodotSharp/GodotSharp/Core/Vector3.cs

```diff
@@ -751,6 +768,50 @@ public readonly Vector3 Snapped(real_t step)
             );
         }
 
+        /// <summary>
+        /// Returns the octahedral-encoded (oct32) form of this Vector3 as a Vector2. Since a Vector2 occupies 1/3 less memory compared to Vector3,
+        /// this form of compression can be used to pass greater amounts of normalized Vector3s without increasing storage or memory requirements.
+        /// See also <see cref="Normalized()"/>, <see cref="OctahedronDecode(Vector2)"/>.
+        /// Note: OctahedronEncode can only be used for normalized vectors. OctahedronEncode does not check whether this Vector3 is normalized,
+        /// and will return a value that does not decompress to the original value if the Vector3 is not normalized.
+		/// Note: Octahedral compression is lossy, although visual differences are rarely perceptible in real world scenarios.
+        /// </summary>
+        /// <returns>The encoded Vector2.</returns>
+        public readonly Vector2 OctahedronEncode()
+        {
+            Vector3 n = this;
+            n /= Math.Abs(n.X) + Math.Abs(n.Y) + Math.Abs(n.Z);
```

Comment:
Even though `Math.Abs` has both a `float` and `double` overload, most methods in `Math` only take `double`. That's why in the Vector types we use `Mathf` everywhere, to ensure both overloads exist and it will work for `real_t` regardless of whether it's `float` or `double`.

```suggestion
            n /= Mathf.Abs(n.X) + Mathf.Abs(n.Y) + Mathf.Abs(n.Z);
```

Reply:
Ah gotcha. For some reason I thought the other calls were to `System.MathF` rather than `Godot.Mathf` 🤦